xmtp/src/lib.rs

```diff
@@ -28,12 +30,25 @@ pub trait Errorer {
 }
 
 pub trait Store<I> {
-    fn store(&self, into: &mut I) -> Result<(), String>;
+    fn store(&self, into: &mut I) -> Result<(), StorageError>;
+}
+
+pub trait KeyStore: Default + Errorer {
+    fn get_account(&mut self) -> Result<Option<Account>, StorageError>;
+
+    fn set_account(&mut self, account: &Account) -> Result<(), StorageError>;
+}
```

Comment:
Adds trait, as the previous approach led to type issues

Reply:
FWIW, I am using the store and fetch traits in my session PR. I like them, and can see the utility for the most boring methods that have the same semantics for every model. I'll be adding some more specific methods too

Reply:
@richardhuaaa The type issues stem from Client not knowing that Store<EncryptedStore> has been implemented for Account. It cannot reasonably know this unless EncryptedStore is in scope which defeats the purpose of making it abstract. 

Either use a trait to define expected behavior, or remove the Generic and make the client aware of EncryptedStore

You are right that I'm still juggling two potential paths while the direction is decided